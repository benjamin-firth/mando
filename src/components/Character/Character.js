import React from 'react';
import logo from '../../logo.svg';
import Loading from '../Loading/Loading.js';

const Character = ({ homeworld, name, population, species, films, updateFave }) => {
  if (!name) {
    return <Loading />
  }
  let filmList = films.map(film => <li>{film}</li>)
  return (
    <article className="movie">
      <div>
        <h2>{name}</h2>
        <p>Homeworld: {homeworld}</p>
        <p>Homeworld Population: {population}</p>
        <p>Species: {species}</p>
        <ul>Films: {filmList}</ul>
      </div>
      <img src={logo} className="movie-picture" alt="Movie Poster" />
      <button onClick={() => updateFave({homeworld, name, population, species, films})}>Fave Dis</button>
    </article>
  )
}

export default Character;
