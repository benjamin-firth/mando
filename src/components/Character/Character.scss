.movie {
  display: flex;
  justify-content: space-between;
  margin: auto;
  margin-top: 20px;
  width: 40vw;
  height: auto;
  background-color: $secondary-background-color;
  div {
    width: 50vw;
    text-align: left;
    .character-button {
      background-color: black;
      color: white;
      width: 40vw;
      font-weight: bold;
      display: inline-block;
      height: 5vh;
    }
  }
  .movie-picture {
    height: inherit;
    width: 30vw;
    object-fit: cover;
    border-left: 2px solid yellow;
  }
}
