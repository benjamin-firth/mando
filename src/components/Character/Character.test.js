import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Character />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot when there is a name', () => {
    let mockProps = { 
      homeworld: 'test earth',
      name: 'John Adams',
      population: 300,
      species: 'tree frog',
      films: ['test', 'test'],
      updateFave: jest.fn(),
    }
    wrapper = shallow(<Character {...mockProps} key='test'/>)
    expect(wrapper).toMatchSnapshot();
  })
})