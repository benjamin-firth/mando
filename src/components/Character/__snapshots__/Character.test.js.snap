// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character should match the snapshot 1`] = `<Loading />`;

exports[`Character should match the snapshot when there is a name 1`] = `
<article
  className="character-card"
>
  <div>
    <h2>
      John Adams
    </h2>
    <p>
      Homeworld: 
      test earth
    </p>
    <p>
      Homeworld Population: 
      300
    </p>
    <p>
      Species: 
      tree frog
    </p>
    <ul>
      Films: 
      <li>
        test
      </li>
      <li>
        test
      </li>
    </ul>
  </div>
  <button
    onClick={[Function]}
  >
    Fave Dis
  </button>
</article>
`;
